platforms/allwinner-d1/boards/lichee-rv/src/main.rs

```diff
@@ -0,0 +1,55 @@
+#![no_std]
+#![no_main]
+
+use core::{panic::PanicInfo, time::Duration};
+use drivers::Ram;
+use kernel::{Kernel, KernelSettings};
+use embedded_hal::blocking::delay::DelayMs;
+
+const HEAP_SIZE: usize = 384 * 1024 * 1024;
+
+#[link_section = ".aheap.AHEAP"]
+#[used]
+static AHEAP: Ram<HEAP_SIZE> = Ram::new();
+
+#[allow(non_snake_case)]
+#[riscv_rt::entry]
+fn main() -> ! {
+    let k = initialize_kernel().unwrap();
+
+    // k.initialize(async {}).unwrap();
+
+    loop {
+        let tick = k.tick();
+
+        if !tick.has_remaining {
+            let mut delay = riscv::delay::McycleDelay::new(1_080_000_000);
+            delay.delay_ms(1);
+            let _turn = k.timer().force_advance_ticks(1);
```

Comment:
this isn't *really* correct: we delay for 1ms every tick, but we don't consider how long it took to tick the kernel scheduler, which may vary substantially based on how many tasks are scheduled on that tick and how long those tasks take to poll. 

i'm fine with merging this as is, but eventually we'll probably want to use a hardware timestamp instead of delaying?